core-java-modules/core-java-21/src/main/java/com/baeldung/java21/SwitchPattern.java

```diff
@@ -0,0 +1,52 @@
+package com.baeldung.java21;
+
+public class SwitchPattern {
+    
+    class Account{
+        double getBalance() {
+            return 0;
+        }
+    }
+    
+    class SavingsAccount extends Account {
+        double getBalance() {
+            return 100;
+        } 
+    }
+    class TermAccount extends Account {
+        double getBalance() {
+            return 1000;
+        } 
+    }
+    class CurrentAccount extends Account {
+        double getBalance() {
+            return 10000;
+        } 
+    }
```

Comment:
```suggestion
    static class Account{
        double getBalance() {
            return 0;
        }
    }
    
    static class SavingsAccount extends Account {
        @Override
        double getBalance() {
            return 100;
        } 
    }
    static class TermAccount extends Account {
        @Override
        double getBalance() {
            return 1000;
        } 
    }
    static class CurrentAccount extends Account {
        @Override
        double getBalance() {
            return 10000;
        } 
    }
```

Reply:
Moved to static class type and test case is also updated.